test/integration/validation/certificaterequest_test.go

```diff
@@ -112,10 +112,11 @@ func TestValidationCertificateRequests(t *testing.T) {
 					Request: mustGenerateCSR(t, &cmapi.Certificate{
 						Spec: cmapi.CertificateSpec{
 							DNSNames: []string{"example.com"},
-							Usages:   []cmapi.KeyUsage{},
+							// mustGenerateCSR will default the usages to digital signature and key encipherment
+							Usages: []cmapi.KeyUsage{},
 						},
 					}),
-					Usages:    []cmapi.KeyUsage{cmapi.UsageDigitalSignature, cmapi.UsageKeyEncipherment, cmapi.UsageClientAuth},
```

Comment:
I don't know why this was not triggering validation errors before, `cmapi.UsageClientAuth` is not a defaulted anywhere.

Reply:
@munnerz do you know why? Also, the meaning of 'special usages' is not 100% clear to me.

Reply:
I feel like we need to know what was going on here before we can merge this PR, no?

Reply:
Agreed, I think the new behavior is correct, but I should at least be able to explain why it was different before.
I'll do some research.

Reply:
I think I found the problem: we only compare the extended key usages (in this case ClientAuth) in case the extension is present in the CertificateRequest. However, if we create a CSR in go and pass it an empty array of extended key usages, the extension is omitted in the CSR blob, which means it is not checked.

Reply:
It seems like we want to support the use case in which usages are only defined in CSR or in CR: https://github.com/cert-manager/cert-manager/pull/3211#discussion_r477241684

Reply:
I created a PR to address this issue: https://github.com/cert-manager/cert-manager/pull/6182